test/widgets/content_test.dart

```diff
@@ -471,6 +471,20 @@ void main() {
       debugNetworkImageHttpClientProvider = null;
     });
 
+   testWidgets('smoke with absolute URL and size(pixelratio * 50)', (tester) async {
+      const avatarUrl = 'https://example/avatar.png';
+      check(await actualUrl(tester, avatarUrl, 50)).isNotNull()
+        .asString.equals(avatarUrl.replaceAll('.png', '-medium.png'));
```

Comment:
The fact that this produces the "medium" URL is dependent on (as your description string suggests) the size in physical pixels, and therefore on what the device pixel ratio is.

There is some default value that tests use for the device pixel ratio, but I don't know offhand what it is. Neither will other likely readers of this test case.

In general, a test case should be self-contained in expressing the key pieces of data that are essential to the test behaving correctly and testing the thing it's meant to test. So this test should explicitly set the device pixel ratio.

Somewhere Flutter will provide a way to do that in tests — might be a method on `tester.binding`, or might be a global with a name starting with "debug", like the `debugNetworkImageHttpClientProvider` you see in the existing tests here. So you should find that and use it. If after looking around the docs you're not finding it, ask in `#mobile-team` and mention where you've looked so far, and we can help look.

Reply:
[f74e3d4](https://github.com/zulip/zulip-flutter/pull/534/commits/f74e3d4e53efee364cad4f2180351fbf72d01b02)